### ISSUE TITLE
Typo in mlptrain/configurations/calculate.py

### ISSUE THREAD

**Issue by tanoury1** [duartegroup/mlp-train#53]
Hi, 
line 24 of calculate.py, "from autode.calculation import Calculation" should be "from autode.calculations import Calculation"

After I fixed that, I'm getting an Attribute error for get_gradients:  

2023-08-01 11:50:08 hpchead2 autode.log.log[106774] INFO Setting output file lines
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/cluster/home/tanoury/miniconda3/envs/mlp-train/lib/python3.9/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/cluster/home/tanoury/mlp-train/mlptrain/configurations/configuration_set.py", line 645, in _single_point_eval
    config.single_point(method_name, **kwargs)
  File "/cluster/home/tanoury/mlp-train/mlptrain/configurations/configuration.py", line 168, in single_point
    run_autode(self, method, n_cores=n_cores)
  File "/cluster/home/tanoury/mlp-train/mlptrain/utils.py", line 49, in wrapped_function
    out = func(*args, **kwargs)
  File "/cluster/home/tanoury/mlp-train/mlptrain/configurations/calculate.py", line 41, in run_autode
    configuration.forces.true = -calc.get_gradients.to('eV Å^-1')
AttributeError: 'Calculation' object has no attribute 'get_gradients'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/cluster/home/tanoury/mlp-train/examples/methane.py", line 18, in <module>
    gap.al_train(method_name='orca', temp=1000)
  File "/cluster/home/tanoury/mlp-train/mlptrain/potentials/_base.py", line 191, in al_train
    al_train(self, method_name=method_name, **kwargs)
  File "/cluster/home/tanoury/mlp-train/mlptrain/training/active.py", line 140, in train
    _gen_and_set_init_training_configs(mlp=mlp,
  File "/cluster/home/tanoury/mlp-train/mlptrain/training/active.py", line 490, in _gen_and_set_init_training_configs
    init_configs.single_point(method_name)
  File "/cluster/home/tanoury/mlp-train/mlptrain/configurations/configuration_set.py", line 414, in single_point
    return self._run_parallel_method(function=_single_point_eval,
  File "/cluster/home/tanoury/mlp-train/mlptrain/configurations/configuration_set.py", line 622, in _run_parallel_method
    self[i] = result.get(timeout=None)
  File "/cluster/home/tanoury/miniconda3/envs/mlp-train/lib/python3.9/multiprocessing/pool.py", line 771, in get
    raise self._value
AttributeError: 'Calculation' object has no attribute 'get_gradients'

Best regards,
Jerry

**Comment by t-young31** [comment]
Hi @tanoury1 – I think you might be using the newly released version of [autodE](https://github.com/duartegroup/autodE) (v1.4.0) which isn't compatible with mlp-train. Did you install mlp-train using the [install script](https://github.com/duartegroup/mlp-train/blob/main/install.sh)? because it should install a pinned version: https://github.com/duartegroup/mlp-train/blob/f42a476e21a4b84422dffdfc1382eebf04cf2a42/requirements.txt#L4

**Comment by tanoury1** [comment]
Thanks t-young31.  I used the install script, but then updated autoDE.  I'll go back to version 1.1*.

**Comment by tanoury1** [comment]
Hi t-young31.  I uninstalled autoDE 1.4, changed calculate.py back to its original, then ran 'python methane.py' from the examples directory.  Also, I set a path to Orca 5.0.  Almost there, I got the following error from quippy/potential.py:

2023-08-01 12:57:04 hpchead2 mlptrain.log[150028] INFO Running 400 steps with a timestep of 0.5 fs
Traceback (most recent call last):
  File "/cluster/home/tanoury/mlp-train/examples/methane.py", line 21, in <module>
    trajectory = mlt.md.run_mlp_md(configuration=system.random_configuration(),
  File "/cluster/home/tanoury/mlp-train/mlptrain/sampling/md.py", line 141, in run_mlp_md
    traj = _run_mlp_md_decorated(configuration=configuration,
  File "/cluster/home/tanoury/mlp-train/mlptrain/utils.py", line 49, in wrapped_function
    out = func(*args, **kwargs)
  File "/cluster/home/tanoury/mlp-train/mlptrain/sampling/md.py", line 226, in _run_mlp_md
    _run_dynamics(ase_atoms=ase_atoms,
  File "/cluster/home/tanoury/mlp-train/mlptrain/sampling/md.py", line 350, in _run_dynamics
    dyn.run(steps=n_steps)
  File "/cluster/home/tanoury/miniconda3/envs/mlp-train/lib/python3.9/site-packages/ase/md/md.py", line 137, in run
    return Dynamics.run(self)
  File "/cluster/home/tanoury/miniconda3/envs/mlp-train/lib/python3.9/site-packages/ase/optimize/optimize.py", line 158, in run
    for converged in Dynamics.irun(self):
  File "/cluster/home/tanoury/miniconda3/envs/mlp-train/lib/python3.9/site-packages/ase/optimize/optimize.py", line 131, in irun
    self.call_observers()
  File "/cluster/home/tanoury/miniconda3/envs/mlp-train/lib/python3.9/site-packages/ase/optimize/optimize.py", line 110, in call_observers
    function(*args, **kwargs)
  File "/cluster/home/tanoury/miniconda3/envs/mlp-train/lib/python3.9/site-packages/ase/io/trajectory.py", line 133, in write
    self._write_atoms(image, **kwargs)
  File "/cluster/home/tanoury/miniconda3/envs/mlp-train/lib/python3.9/site-packages/ase/io/trajectory.py", line 168, in _write_atoms
    c.write(name=calc.name)
  File "/cluster/home/tanoury/miniconda3/envs/mlp-train/lib/python3.9/site-packages/quippy/potential.py", line 121, in name
    return self._get_name()
  File "/cluster/home/tanoury/miniconda3/envs/mlp-train/lib/python3.9/site-packages/quippy/potential.py", line 117, in _get_name
    return self.name_
AttributeError: 'Potential' object has no attribute 'name_'

We are close.

Best regards,
Jerry

**Comment by t-young31** [comment]
Could be wrong but it might be that quippy isn't compatible with the as yet unreleased version of ase (required for `ase.calculators.plumed`). Merging #28 has probably broken the examples. @vitartas – got any thoughts here?

**Comment by vitartas** [comment]
We have not used GAP for a long time now, so not exactly sure. In https://github.com/libAtoms/QUIP/blob/public/quippy/quippy/potential.py, there is no `get_name()`, which suggests that updating quippy might fix the issue. However, I will try to make a fresh GAP environment to see if the examples work properly with the latest version of _mlp-train_.

**Comment by tanoury1** [comment]
Thanks for updating the code for GAP.  I'll give it a go later today/tomorrow.  Moved on to water.py example using ACE, since ACE is more efficient and accurate than GAP.  Have some questions around that:
1. Here's how I did it: initially install mlptrain via a conda environment 'mlp-train'
2. Installed ACE by sourcing install_ace.py.  Accordingly, the new 'ace' conda environment should be able to run the entire water.py job.
3.   After 'python water.py', I get the following error:
(ace) [tanoury@hpchead2:examples]$ python water.py 
Traceback (most recent call last):
  File "water.py", line 1, in <module>
    import mlptrain as mlt
  File "/cluster/home/tanoury/mlp-train/mlptrain/__init__.py", line 6, in <module>
    from mlptrain.sampling import md, UmbrellaSampling, Metadynamics
  File "/cluster/home/tanoury/mlp-train/mlptrain/sampling/__init__.py", line 2, in <module>
    from mlptrain.sampling.plumed import PlumedBias, PlumedCalculator
  File "/cluster/home/tanoury/mlp-train/mlptrain/sampling/plumed.py", line 7, in <module>
    from ase.calculators.plumed import Plumed
ModuleNotFoundError: No module named 'ase.calculators.plumed'

Best regards,
Jerry
P.S. Should I make this a new issue

**Comment by tanoury1** [comment]
Made more progress here.  My initial issue was that I installed ACE while I was in my mlp-train conda environment.  Removed my ace environment, then created another one (while not in mlp-train).  Additionally, needed to 'source install_gap.sh' because water.py was looking for xtb, which is installed in install_gap.sh only.  

Got a julia error which I do not understand:
2023-08-02 11:02:37 hpchead2 autode.log.log[168547] INFO Getting gradients from tmp_xtb.out
2023-08-02 11:02:37 hpchead2 autode.log.log[168547] INFO Getting energy from tmp_xtb.out
2023-08-02 11:02:37 hpchead2 autode.log.log[168547] INFO Checking for tmp_xtb.out normal termination
2023-08-02 11:02:37 hpchead2 autode.log.log[168547] INFO Getting atomic charges from tmp_xtb.out
2023-08-02 11:02:37 hpchead2 mlptrain.log[168547] WARNING Not adding basis functions on H
2023-08-02 11:02:37 hpchead2 mlptrain.log[168547] WARNING Save called without defining what energy and forces to print. Had true energies to using those
Failed to find a Julia installation. Make sure it's present in your $PATH

Julia was installed via 'pip install julia --user' during the ACE build.  Do I need to explicitly have a path to this installation of Julia in $PATH, even though is was installed as part of the ace conda environment?

Jerry

**Comment by tanoury1** [comment]
The saga continues.  I set a path to the bin of own Julia install (julia-1.7.3).  water.py then gave the error:
Traceback (most recent call last):
  File "water.py", line 11, in <module>
    ace.al_train(method_name='xtb', temp=500)
  File "/cluster/home/tanoury/mlp-train/mlptrain/potentials/_base.py", line 191, in al_train
    al_train(self, method_name=method_name, **kwargs)
  File "/cluster/home/tanoury/mlp-train/mlptrain/training/active.py", line 157, in train
    mlp.train()
  File "/cluster/home/tanoury/mlp-train/mlptrain/potentials/_base.py", line 63, in train
    self._train()
  File "/cluster/home/tanoury/mlp-train/mlptrain/potentials/ace/ace.py", line 51, in _train
    raise RuntimeError(f'ACE train errored with:\n{err.decode()}\n')
RuntimeError: ACE train errored with:
ERROR: LoadError: ArgumentError: Package IPFitting not found in current path:
- Run `import Pkg; Pkg.add("IPFitting")` to install the IPFitting package.

Stacktrace:
 [1] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:967
in expression starting at /cluster/home/tanoury/mlp-train/examples/water.jl:1

I tried to import the IPFitting package into my julia-1.7.3 install, but julia says:
IPFitting (not found in project, manifest or registry)

Best regards,
Jerry

**Comment by tanoury1** [comment]
Slowly moving along here.  I set my PATH to include my julia-1.7.3 install.  Per ACE.jl documentation, I needed to 
registry add https://github.com/JuliaMolSim/MolSim.git to be able to add IPFitting
also had to add PyCall, ACE, JuLIP, ASE and ACEatoms.

Now, getting an z2i error:
Traceback (most recent call last):
  File "water.py", line 11, in <module>
    ace.al_train(method_name='xtb', temp=500)
  File "/cluster/home/tanoury/mlp-train/mlptrain/potentials/_base.py", line 191, in al_train
    al_train(self, method_name=method_name, **kwargs)
  File "/cluster/home/tanoury/mlp-train/mlptrain/training/active.py", line 157, in train
    mlp.train()
  File "/cluster/home/tanoury/mlp-train/mlptrain/potentials/_base.py", line 63, in train
    self._train()
  File "/cluster/home/tanoury/mlp-train/mlptrain/potentials/ace/ace.py", line 51, in _train
    raise RuntimeError(f'ACE train errored with:\n{err.decode()}\n')
RuntimeError: ACE train errored with:
ERROR: LoadError: UndefVarError: z2i not defined
in expression starting at /cluster/home/tanoury/mlp-train/examples/water.jl:3

Jerry

**Comment by juraskov** [comment]
Dear Jerry, thanks for reporting all this. We have completely rewritten the installation scripts to (hopefully) avoid the problems you had. The current version checks for the existence of Julia installation and returns errors when not found. I am now closing the issue.

**Comment by tanoury1** [comment]
Thank you.  I'll update and give it a go.

**Comment by physorgchem** [comment]
@juraskov thanks for the update! It was much easier to install however, I had a problem with IPFitting.
What is the recommended Julia version? I have tried 1.7.1 and 1.9.3 and both had problems with IPFitting.

`Precompiling project...
  ? IPFitting
  1 dependency successfully precompiled in 15 seconds (169 already precompiled)
  1 dependency failed but may be precompilable after restarting julia
* DONE! *`

I have tried to manually install IPFitting but could not solve this problem. 

Any suggestions? Thanks!
